### PR TITLE
[FIX] CORS error while interacting with any action button on Livechat

### LINKED ISSUE #21033: Enable CORS in ui.interaction endpoint
### Description:

Hi @sampaiodiego ! I hope you can help me with this.

I have an issue implementing the UI Kit library.

I have an error calling to `api/apps/ui.interaction/c33fa1a6-68a7-491e-bf49-9d7b99671c48` endpoint: 
```
Access to XMLHttpRequest at 'http://chat.local:3300/api/apps/ui.interaction/c33fa1a6-68a7-491e-bf49-9d7b99671c48' from origin 'http://localhost:8000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource
```

I see I can add cors easily into the code: 
https://github.com/RocketChat/Rocket.Chat/blob/e802fa59ce48fdee93a5ef56136b076acb213d87/app/apps/server/communication/uikit.js#L13
Whit this code:
https://expressjs.com/en/resources/middleware/cors.html

Also I see in the configuration of the RC Server that I can enable or disable the CORS in REST API, but that setting does not apply in this `ui.interaction` endpoint, right?

I would like to enable the CORS as same as REST API configuration.
I can make a PR, but I don't know how to get such configuration from `uikit.js` file. Can you point me in the right direction?

Thanks a lot!

### Steps to reproduce:

1. Enable CORS in REST Configuration.
2. I can query to `http://chat.local:3300/api/v1/method.call/slashCommand` but not to  `http://chat.local:3300/api/apps/ui.interaction/c33fa1a6-68a7-491e-bf49-9d7b99671c48`

### Expected behavior:

To works with CORS enable as same as REST API `api/v1/...`

### Actual behavior:

image

### Server Setup Information:

Server Setup Information
Versión
3.11.1
ID de despliegue
vcohFpCn68jhkCkHT
Versión de Motor de Aplicaciones
1.22.2
Versión de Node
v12.18.4
Migración de base de datos
215 (7 de marzo de 2021 10:47
MongoDB
4.0.23 / mmapv1 (oplog Habilitado)
Detalles de la transacción
HEAD: (7c5d6ae2a)
Rama: HEAD
PID
10

### Version of Rocket.Chat Server

3.11.1

### Client Setup Information

- Desktop App or Browser Version: Chrome. Custom client app.
- Operating System: LINUX

### Additional context

I am developing a custom client to integrate with RC Server. And we want to connect with Poll app. So I need to implement the UI Kit.

### PR DESCRIPTION
**Pull request by murtaza98**
## Issue(s)

#21033

### CODE REVIEW

**Review by lolimay — DISMISSED** [review]

File: app/apps/server/communication/uikit.js [hunk 1]
```diff
@@ -14,6 +15,25 @@ const apiServer = express();
 
 apiServer.disable('x-powered-by');
 
+if (settings.get('API_Enable_CORS')) {
+	const CORSOriginSetting = settings.get('API_CORS_Origin');
+
+	const whitelistOrigins = CORSOriginSetting
```
Comment (d-gubert) [inline app/apps/server/communication/uikit.js:21]: I'd suggest not using `blacklist/whitelist` names  and instead using the more "modern" approach to naming it as `blocklist/allowlist` 😉

**Comment by d-gubert** [comment]
It looks good to me, I'd just like to get the eyes of @sampaiodiego to check on the use of the Settings 🙂

**Comment by murtaza98** [comment]
> the only issue I'm seeing is that if the settings are updated the new values are not read, so it would require a server restart to get updates CORS values.

Thanks this was helpful! :smile: Done with this change

**Review by sampaiodiego — CHANGES_REQUESTED** [review]
Thanks @murtaza98 the code looks better now, but while testing I was getting "Not allowed by CORS" even with CORS disabled and using from same domain (using from rocket.chat web client actually).

can you take a look on that pls? thx

**Comment by murtaza98** [comment]
> Thanks @murtaza98 the code looks better now, but while testing I was getting "Not allowed by CORS" even with CORS disabled and using from same domain (using from rocket.chat web client actually).
> 
> can you take a look on that pls? thx

Hey @sampaiodiego I was able to solve this issue by adding a new check there in callback `origin === settings.get('Site_Url')`. Could you please review it again? Thanks :smile:

**Comment by sampaiodiego** [comment]
hi @murtaza98 .. I tested it again and even with CORS disabled I'm not able to use `/poll` due to CORS erros, as you can see on attachments below:

image

image

image

**Comment by murtaza98** [comment]
Hey @sampaiodiego There was an issue within the condition over [here](https://github.com/RocketChat/Rocket.Chat/pull/21406/files#diff-f1739e229241c8ae25c7d00c680d188e45f9552c3a5d9c17c11610191fb385c9R29) due to which you were getting the error. I've now updated the logic to handle this. Request you to please review it again. Thanks!

**Comment by murtaza98** [comment]
Guys, Some of the builds are failing here and the error is not related to this PR. The error was being caused by some of the merge changes I made in this PR. Due to this, I decided to close this PR and create a new PR with only relevant change. 

NEW PR: #22121

**Pull request closed without merging**